src/Components/OATHeader/OATHeader.tsx

```diff
@@ -315,35 +312,78 @@ const OATHeader = ({ dispatch, state }: OATHeaderProps) => {
         reader.readAsDataURL(e.target.files[0]);
     };
 
+    const isJsonValid = (text) => {
+        try {
+            JSON.parse(text);
+        } catch (e) {
+            return false;
+        }
+        return true;
+    };
+
     const handleFileListChanged = async (files: Array<File>) => {
-        const items = [];
+        const newModels = [];
         if (files.length > 0) {
             const filesErrors = [];
             let modelsMetadataReference = null;
             for (const current of files) {
                 const content = await current.text();
-                const error = await parseModel(content);
-                if (!error) {
-                    modelsMetadataReference = populateMetadata(
-                        current,
-                        content,
-                        modelsMetadataReference
+                const validJson = isJsonValid(content);
+                if (validJson) {
+                    const jsonContent = JSON.parse(content);
```

Comment:
Can we avoid parsing the JSON twice? I.e. we parse it once in the `isJsonValid` function and then again here.

Could we do something like a `safeJsonParse` function that just returns null if it's invalid? And if it's null, then add the error message?